scripts/competitiond.py

```diff
@@ -0,0 +1,580 @@
+#!./venv/bin/python
+"""
+Competition leaderboard evaluation daemon.
+
+Does the following in a loop, every {refresh_period_seconds}:
+    1. Find bundles tagged with {submission_tag} and filter them.
+    2. Run the {predict} command with the submitted bundle to generate
+       predictions on the test set.
+    3. Tag the resulting test run bundle with {predict.tag}, untagging
+       any previous test run bundles for the same submitter.
+    4. Run {evaluate} command with the test run bundle.
+    5. Tag the resulting evaluation bundle with {evaluate.tag}, untagging
+       any previous evaluation bundles for the same submitter.
+
+All bundles created by this daemon are added to {log_worksheet_uuid}.
+Each user will be limited to {max_submissions_per_day}.
+
+Example config file:
+
+    {
+        "max_submissions_per_day": 1,
+        "refresh_period_seconds": 60,
+        "host": "https://worksheets.codalab.org",
+        "username": "competition-manager5",
+        "password": "xxxxxxxxxxxx",
+        "submission_tag": "submission-test",
+        "log_worksheet_uuid": "0x08b7241ff2cf4e3eba59ad7ca1b9639f",
+        "predict": {
+            "dependencies": [
+                "testset:0x2fb9723f18844d6ba358af7a67e22b96",
+                "model.py:{submission_uuid}"
+            ],
+            "command": "python model.py testset",
+            "tag": "comp-prediction"
+        },
+        "evaluate": {
+            "dependencies": [
+                "testset:0x2fb9723f18844d6ba358af7a67e22b96",
+                "evaluate.py:0x210dc57046e74957aef56b8b89cb53c7",
+                "predictions.json:{predictions_uuid}"
+            ],
+            "command": "python evaluate.py testset predictions.json",
+            "tag": "comp-leaderboard"
+        }
+    }
+
+The following string substitutions will be made in the dependency specs:
+
+    {submission_uuid} => UUID of the user submitted bundle
+    {predictions_uuid} => UUID of the resulting test run bundle
+
+"""
+import argparse
+import getpass
+import json
+import logging
+import re
+import signal
+import sys
+import time
+import traceback
+from collections import defaultdict
+
+from marshmallow import Schema, fields, ValidationError
+
+sys.path.append('.')
```

Comment:
What's the purpose of this line? How would the behavior be different without it? Is it to avoid using a system version of Codalab? And if so, why is using the system version of Codalab bad?